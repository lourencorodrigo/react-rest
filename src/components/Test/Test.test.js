import React from 'react';
import Test from './Test';
import { shallow, mount } from 'enzyme';
const ReactTestRenderer = require('react-test-renderer');

it('should render without throwing an error', function () {
  expect(shallow(<Test name="Rodrigo" />).contains(
    <h1>Hello, Rodrigo!</h1>
  )).toBe(true);
});

it('should mount in a full DOM', function() {
  expect(mount(<Test name="Rodrigo" />).find('h1').length).toBe(1);
});

it('should render to static HTML', () => {
  const wrapper = shallow(<Test name="Rodrigo" />);
  expect(wrapper.find('h1').text()).toEqual('Hello, Rodrigo!');
});

it('Should compare the component with a snapshot', () => {
  const component = ReactTestRenderer.create(<Test name="Rodrigo" />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});