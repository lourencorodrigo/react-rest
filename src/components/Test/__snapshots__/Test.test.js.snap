// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should compare the component with a snapshot 1`] = `
<h1>
  Hello, Rodrigo!
</h1>
`;
